src/sentry/workflow_engine/handlers/condition/group_state_handlers.py

```diff
@@ -0,0 +1,27 @@
+from typing import Any
+
+from sentry.workflow_engine.models.data_condition import Condition
+from sentry.workflow_engine.registry import condition_handler_registry
+from sentry.workflow_engine.types import DataConditionHandler, WorkflowJob
+
+
+@condition_handler_registry.register(Condition.REGRESSED_EVENT)
+class RegressedEventConditionHandler(DataConditionHandler[WorkflowJob]):
+    @staticmethod
+    def evaluate_value(job: WorkflowJob, comparison: Any) -> bool:
+        state = job.get("group_state", None)
+        if state is None:
+            return False
+
+        return state["is_regression"] == comparison
```

Comment:
is `is_regression` guaranteed to exist on `state`? if not I'd do a `.get()`

Reply:
yes it's guaranteed, the state is a typed dict too https://github.com/getsentry/sentry/blob/a62dc3bceb196b2caa327e2941bad15980d717cc/src/sentry/eventstream/base.py#L30-L34